profiler/src/ProfilerEngine/Datadog.Profiler.Native/DebugInfoStore.cpp

```diff
@@ -0,0 +1,158 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+#include "DebugInfoStore.h"
+
+#include "COMHelpers.h"
+#include "Log.h"
+#include "IConfiguration.h"
+
+#include "shared/src/native-src/string.h"
+
+#define GUID_DEFINED
+#define MDTOKEN_DEFINED
+#include "shared/src/native-lib/PPDB/inc/PPDBReader.hpp"
+#undef MDTOKEN_DEFINED
+#undef GUID_DEFINED
+
+const std::string DebugInfoStore::NoFileFound = "";
+const std::uint32_t DebugInfoStore::NoStartLine = 0;
+
+DebugInfoStore::DebugInfoStore(ICorProfilerInfo4* profilerInfo, IConfiguration* _configuration) :
+    _profilerInfo{profilerInfo},
+    _isEnabled{_configuration->IsDebugInfoEnabled()}
+{
+}
+
+SymbolDebugInfo DebugInfoStore::Get(ModuleID moduleId, mdMethodDef methodDef)
+{
+    if (!_isEnabled)
+    {
+        return {NoFileFound, NoStartLine};
+    }
+
+    std::unique_lock _l(_modulesMutex);
+
+    auto rid = RidFromToken(methodDef);
+    auto it = _modulesInfo.find(moduleId);
+    if (it == _modulesInfo.cend())
+    {
+        ParseModuleDebugInfo(moduleId);
+    }
+
+    auto const& info = _modulesInfo[moduleId];
```

Comment:
Still tricky, because if it's not in the cache we need to lookup again.
But we may change the logic a bit here. I'll see
> We already seen that lookup are consuming CPU especially in hot path

I do not recall that. The lookup here is hashing a uint64_t....